pkg/connector/handlewhatsapp.go

```diff
@@ -0,0 +1,120 @@
+package connector
+
+import (
+	"context"
+	"fmt"
+
+	"go.mau.fi/whatsmeow/types"
+	"go.mau.fi/whatsmeow/types/events"
+	"maunium.net/go/mautrix/bridge/status"
+	"maunium.net/go/mautrix/bridgev2"
+	"maunium.net/go/mautrix/bridgev2/networkid"
+	"maunium.net/go/mautrix/bridgev2/simplevent"
+
+	"maunium.net/go/mautrix-whatsapp/pkg/waid"
+)
+
+const (
+	WADisconnected   status.BridgeStateErrorCode = "wa-transient-disconnect"
+	WAPermanentError status.BridgeStateErrorCode = "wa-unknown-permanent-error"
+)
+
+func (wa *WhatsAppClient) handleWAEvent(rawEvt any) {
+	log := wa.UserLogin.Log
+
+	switch evt := rawEvt.(type) {
+	case *events.Message:
+		log.Trace().
+			Any("info", evt.Info).
+			Any("payload", evt.Message).
+			Msg("Received WhatsApp message")
+
+		portalKey, ok := wa.makeWAPortalKey(evt.Info.Chat)
+		if !ok {
+			log.Warn().Stringer("chat", evt.Info.Chat).Msg("Ignoring WhatsApp message with unknown chat JID")
+			return
+		}
+		wa.Main.Bridge.QueueRemoteEvent(wa.UserLogin, &WAMessageEvent{
+			Message:   evt,
+			portalKey: portalKey,
+			wa:        wa,
+		})
+	case *events.Receipt:
+		portalKey, ok := wa.makeWAPortalKey(evt.Chat)
+		if !ok {
+			log.Warn().Stringer("chat", evt.Chat).Msg("Ignoring WhatsApp receipt with unknown chat JID")
+			return
+		}
+		var evtType bridgev2.RemoteEventType
+		switch evt.Type {
+		case types.ReceiptTypeRead, types.ReceiptTypeReadSelf:
+			evtType = bridgev2.RemoteEventReadReceipt
+		case types.ReceiptTypeDelivered:
+			evtType = bridgev2.RemoteEventDeliveryReceipt
+		}
```

Comment:
this might be more clear

```suggestion
		evtType := bridgev2.RemoteEventReadReceipt
		if evt.Type == types.ReceiptTypeDelivered {
			evtType = bridgev2.RemoteEventDeliveryReceipt
		}
```

Reply:
There are other types of receipts too which are missing here, but a switch will be better once they're added

Reply:
yeah I removed the unsupported ones

Reply:
makes sense, ignore me then lol